app/uk/gov/hmrc/cgtpropertydisposals/repositories/CacheRepository.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.repositories
+
+import com.google.inject.Singleton
+import play.api.libs.json._
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.{BSONDocument, BSONObjectID}
+import reactivemongo.play.json.collection.JSONBatchCommands.FindAndModifyCommand
+import uk.gov.hmrc.cgtpropertydisposals.models.Error
+import uk.gov.hmrc.mongo.ReactiveRepository
+import reactivemongo.play.json.ImplicitBSONHandlers.JsObjectDocumentWriter
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats.dateTimeWrite
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.control.NonFatal
+
+@Singleton
+trait CacheRepository[A] { this: ReactiveRepository[A, BSONObjectID] =>
+
+  val cacheTtl: Int
```

Comment:
it'd be easier to work with if this was a `FiniteDuration`. Then you could convert it to whatever units you need whenever you need it